burn-core/src/module/param/running.rs

```diff
@@ -1,14 +1,33 @@
+use alloc::{string::ToString, sync::Arc, vec, vec::Vec};
+
 use super::{load_with_id, state_with_id};
 use crate::module::{LoadingError, Module, ModuleVisitor, ModuleVisitorMut, Param, State};
 use burn_tensor::{
     backend::{ADBackend, Backend},
     Data, Tensor,
 };
-use std::{
-    collections::HashMap,
-    sync::{Arc, Mutex, RwLock},
-    thread::ThreadId,
-};
+
+cfg_if::cfg_if! {
```

Comment:
Is it the only place where `cfg_if` is used? I'm not a bit fan, it can easily be written without it.

```rust
#[cfg(feature = "std")] 
mod threading {
       use std::thread::ThreadId;
       use std::sync::{Mutex, RwLock};
       use std::collections::HashMap;

       #[inline(always)]
       fn get_thread_current_id() -> ThreadId {
           std::thread::current().id()
      }
}

#[cfg(not(feature = "std"))]
mod threading {
       use burn_common::stub::{Mutex, RwLock, ThreadId};
       use hashbrown::HashMap;

       #[inline(always)]
       fn get_thread_current_id() -> ThreadId {
           panic!("Current thread id is not available")
       }
}
use threading::*;
```

Reply:
Here is a the final version to record. The import's visibility needed pub(super) visibility. 

```rust

#[cfg(feature = "std")]
mod threading {
    pub(super) use std::collections::HashMap;
    pub(super) use std::sync::{Mutex, RwLock};
    pub(super) use std::thread::ThreadId;

    #[inline(always)]
    pub(super) fn get_thread_current_id() -> ThreadId {
        std::thread::current().id()
    }
}

#[cfg(not(feature = "std"))]
mod threading {
    pub(super) use burn_common::stub::{Mutex, RwLock, ThreadId};
    pub(super) use hashbrown::HashMap;

    #[inline(always)]
    pub(super) fn get_thread_current_id() -> ThreadId {
        panic!("Current thread id is not available")
    }
}

```

Reply:
I like this approach because my vscode's rust-analyzer disables blocks if cfg yields false like this:

<img width="985" alt="image" src="https://user-images.githubusercontent.com/939125/221299013-63c8c1af-5946-4979-9388-b31130ae049a.png">